src/Host.cpp

```diff
@@ -225,59 +230,478 @@ void Host::saveModules(int sync)
     if (!savePath.exists()) {
         savePath.mkpath(dirName);
     }
+    QString dateTimeStampString = QDateTime::currentDateTime().toString("dd-MM-yyyy#hh-mm-ss");
+
     while (it.hasNext()) {
         it.next();
-        QStringList entry = it.value();
-        QString filename_xml = entry[0];
+        QStringList moduleData = it.value();
+        QString filename_xml = moduleData.at(0);
         // CHECKME: Consider changing datetime spec to more "sortable" "yyyy-MM-dd#hh-mm-ss" (1 of 6)
-        QString time = QDateTime::currentDateTime().toString("dd-MM-yyyy#hh-mm-ss");
         QString moduleName = it.key();
         QString zipName;
-        zip* zipFile = nullptr;
+        zip* archive = nullptr;
         // Filename extension tests should be case insensitive to work on MacOS Platforms...! - Slysven
-        if (filename_xml.endsWith(QStringLiteral("mpackage"), Qt::CaseInsensitive) || filename_xml.endsWith(QStringLiteral("zip"), Qt::CaseInsensitive)) {
-            QString packagePathName = mudlet::getMudletPath(mudlet::profilePackagePath, mHostName, moduleName);
-            filename_xml = mudlet::getMudletPath(mudlet::profilePackagePathFileName, mHostName, moduleName);
-            int err;
-            zipFile = zip_open(entry[0].toStdString().c_str(), 0, &err);
-            zipName = filename_xml;
-            QDir packageDir = QDir(packagePathName);
-            if (!packageDir.exists()) {
-                packageDir.mkpath(packagePathName);
+
+#if defined(LIBZIP_VERSION_MAJOR) && (LIBZIP_VERSION_MAJOR >= 1)
+        {
+           /*
+            * Code for libzip 1.0 or later - blocked to allow code folding in IDE
+            */
+            if (filename_xml.endsWith(QStringLiteral(".mpackage"), Qt::CaseInsensitive)
+              ||filename_xml.endsWith(QStringLiteral(".zip"), Qt::CaseInsensitive)) {
+                // This branch will overwrite THE EXISTING MODULE FILE WITH NO BACKUP
+                // It can also cause DATA LOSS should MORE THAN ONE PROFILE HAVE
+                // modified the MODULE and it not be
```

Comment:
Unfinished comment?

Reply:
Yeah - I'm not entirely sure that if __more than one__ active profile has modified a module/package and __more then one of them__ are set to be __synced__ which one will destroy/lose/overwrite the changes in the other ones.  If a module/package is not set to be synced it doesn't matter - any changes to those will automatically be thrown away when the profile is close but having the __same__ module/package being set to be synced on _active_ profiles is going to be risky... :bomb: